src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -171,10 +176,12 @@ def __init__(
                 name=name,
                 tupleize_cols=tupleize_cols,
             )
+        # Set the parent of the Index object - used with the name APIs
+        self._parent = data if isinstance(data, BasePandasDataset) else None
```

Comment:
you can do `data = data._query_compiler` here to avoid the update on `set_query_compiler`

Reply:
done